app/controllers/events_controller.rb

```diff
@@ -0,0 +1,55 @@
+class EventsController < ApplicationController
+  after_action :invite_creator, only: [:create]
+  def index
+    @events = Event.all.order('created_at DESC')
+    @upcoming_events = Event.upcoming_events
+    @prev_events = Event.prev_events
+
+    redirect_to new_session_path unless signed_in?
```

Comment:
- [ ] **[OPTIONAL]** We can use this as a method in before_action to restrict access to index action. This way it will be more reusable if wanted to restrict access in another action or in another controller.